boards/arm/nucleo_f207zg/doc/index.rst

```diff
@@ -139,6 +141,7 @@ Default Zephyr Peripheral Mapping:
 
 - UART_3 TX/RX : PD8/PD9 (ST-Link Virtual Port Com)
 - UART_6 TX/RX : PG14/PG9 (Arduino Serial)
+- SPI1 NSS/SCK/MISO/MOSI : PA4/PA5/PA6/PA7 (Arduino SPI)
```

Comment:
Please also update `boards/arm/nucleo_f207zg/arduino_r3_connector.dtsi` to configure arduino_spi in dts:
`arduino_spi: &spi1 {};`
When doing so please also update `supported` section in boards yaml to document it supports `arduino_spi`.